File: 2020/javascript/02/01.js
//----Load file content
fs = require('fs');
var input = fs.readFileSync('input', 'utf8');
var inputArray = input.split('\n');
//---end load

var example = inputArray[0];

var splitLine = inputArray[0].split(':');

var password = splitLine[1];

var policyRules = splitLine[0];

var policyLetter = policyRules.split(' ')[1];

var occurances = 0;

for (var letter in password){
	
	if ( password[letter] == policyLetter ){
		occurances++;
	}
}

File: 2020/python/02/01.py
inputFile = open('input')
lines = inputFile.readlines()

correctPasswords = []

for line in lines:
    splitLine = line.split(':')
    password = splitLine[1] ##password

    policySplit = splitLine[0].split(' ')
    policy = policySplit[1] ##policy

    occuranceseSplit = policySplit[0].split('-')

    minOccur = int(occuranceseSplit[0]) #min
    maxOccur = int(occuranceseSplit[1]) #max

    occurances = 0

    for i in password:
        if i == policy:
            occurances+=1

    if occurances >= minOccur and occurances <= maxOccur:
        correctPasswords.append(line)

#print(correctPasswords)

print(len(correctPasswords))
